shell/renderer/api/electron_api_web_frame.cc

```diff
@@ -904,6 +907,16 @@ class WebFrameRenderer : public gin::Wrappable<WebFrameRenderer>,
 
     return render_frame->GetRoutingID();
   }
+
+  std::string GetWindowFeatures(v8::Isolate* isolate) {
+    content::RenderFrame* render_frame;
+    if (!MaybeGetRenderFrame(isolate, "windowFeatures", &render_frame))
+      return "";
+
+    blink::LocalFrameToken frame_token = render_frame->GetWebFrame()->GetLocalFrameToken();
+    blink::LocalFrame* frame = blink::LocalFrame::FromFrameToken(frame_token);
+    return frame ? frame->GetPage()->GetWindowFeatures().raw_features.Utf8() : "";
```

Comment:
I don't love reaching into the private Blink APIs here. Further, the `GetWindowFeatures` function is only used in one (relatively obscure) place in Blink, so it seems liable to be removed. Can we do this some other way?

Reply:
It's not ideal, the path I used to get the window feature data is the only one I'm aware of.

I believe the code where the window features are saved is a stable area though. See https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/page/create_window.cc;l=352;drc=f51c47ef6c37f12d898997ba00010eb7fba0d090

If the window features were to be removed from the `Page` object, I believe restoring its availability in create_window.cc (linked above) would be straightforward. If we're okay with this bet, I can include a comment above the highlighted code to mention this context for any future upgrade maintainer.

An alternative approach could be to duplicate the `raw_features` string somewhere in blink to then be accessed by our API.